src/middlewared/middlewared/plugins/pool.py

```diff
@@ -778,12 +779,33 @@ async def do_create(self, job, data):
         asyncio.ensure_future(self.restart_services())
 
         pool = await self.get_instance(pool_id)
-        await self.middleware.call_hook('pool.post_create_or_update', pool=pool)
-        await self.middleware.call_hook(
-            'dataset.post_create', {'encrypted': bool(encryption_dict), **encrypted_dataset_data}
-        )
+        if not await self.middleware.call('failover.licensed'):
+            await self.middleware.call_hook('pool.post_create_or_update', pool=pool)
+            await self.middleware.call_hook(
+                'dataset.post_create', {'encrypted': bool(encryption_dict), **encrypted_dataset_data}
+            )
+        else:
+            # we dont have a bulk update/insert method so on large systems where we need to update
+            # the storage_disk table with, potentially, 100's or even 1k+ updates this is painful.
+            # so on HA systems there is no reason to wait for this
+            asyncio.ensure_future(self.middleware.call(
+                'pool.sync_zfs_guid_and_sync_db_to_remote_node',
+                'pool_create', pool, encryption_dict, encrypted_dataset_data
+            ))
+
         return pool
 
+    @private
+    async def sync_zfs_guid_and_sync_db_to_remote_node(self, schema, pool, enc_dict=False, enc_data={}):
+        await self.middleware.call('disk.sync_zfs_guid', pool, False)
+        try:
+            await self.middleware.call('failover.send_small_file', TRUENAS_DB)
```

Comment:
I am not sure you can do that... replacing the entire database underneath middlewared feet on passive node, a lot of hilarity can ensue...
I thought there was a method to re-send the whole database using sqlite?